packages/node_modules/@webex/webex-core/src/lib/services/services.js

```diff
@@ -0,0 +1,374 @@
+   * `ServiceUrl` to be removed from the its
+   * respective host array, and then return the next
+   * viable host from the `ServiceUrls` host array,
+   * or the `ServiceUrls` default url if no other priority
+   * hosts are available, or if `noPriorityHosts` is set to
+   * `true`.
+   * @param {string} url
+   * @param {boolean} noPriorityHosts
+   * @returns {string}
+   */
+  markFailedUrl(url, noPriorityHosts) {
+    const catalog = this._getCatalog();
+
+    return catalog.markFailedUrl(url, noPriorityHosts);
+  },
+
+  /**
+   * Update a list of `serviceUrls` to the most current
+   * catalog via the defined `discoveryUrl` then returns the current
+   * list of services.
+   * @param {object} [param]
+   * @param {string} [param.from] - This accepts `limited` or `signin`
+   * @param {object} [param.query] - This accepts `email`, `orgId` or `userId` key values
+   * @param {string} [param.query.email] - must be a standard-format email
+   * @param {string} [param.query.orgId] - must be an organization id
+   * @param {string} [param.query.userId] - must be a user id
+   * @param {string} [param.token] - used for signin catalog
+   * @returns {Promise<object>}
+   */
+  updateServices({from, query, token} = {}) {
+    const catalog = this._getCatalog();
+    let formattedQuery, serviceGroup;
+
+    // map catalog name to service group name.
+    switch (from) {
+      case 'limited': serviceGroup = 'preauth'; break;
+      case 'signin': serviceGroup = 'signin'; break;
+      default: serviceGroup = 'postauth'; break;
+    }
+
+    // confirm catalog update for group is not in progress.
+    if (catalog.status[serviceGroup].collecting) {
+      return this.waitForCatalog(serviceGroup);
+    }
+
+    catalog.status[serviceGroup].collecting = true;
+
+    // encode email when query key is email
+    if (serviceGroup === 'preauth' || serviceGroup === 'signin') {
+      const queryKey = Object.keys(query)[0];
+
+      formattedQuery = {};
+      formattedQuery[queryKey] = queryKey === 'email' ?
+        base64.encode(query.email) : query[queryKey];
+    }
+
+    return this._fetchNewServiceHostmap({
+      from,
+      token,
+      query: formattedQuery
+    })
+      .then((serviceHostMap) => {
+        catalog.updateServiceUrls(serviceGroup, serviceHostMap);
+        this.updateCredentialsConfig();
+        catalog.status[serviceGroup].collecting = false;
+      })
+      .catch(() => {
+        catalog.status[serviceGroup].collecting = false;
+
+        return Promise.reject();
+      });
+  },
+
+  /**
+   * Validate if a user, by email, is activated or not. This also returns
+   * an object that contains the atlas user details if available.
+   * @param {object} param
+   * @param {string} param.email - must be a standard-format email
+   * @param {string} [param.reqId] - request id
+   * @returns {Promise<object>} - { activated: boolean, details: string, user: object }
+   */
+  validateUser({email, reqId = 'WEBCLIENT'} = {}) {
```

Comment:
Why is the default reqID 'WEBCLIENT'?

Reply:
`WEBCLIENT` is set as the default for `internal-plugin-user`, and `validateUser()` directly replaces the method that uses this data, but can't pull defaults from that plugin.